docs/docs/advanced/setNativeProps.mdx

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_position: 7
+---
+
+# setNativeProps
+
+`setNativeProps` lets you directly update component properties from the [UI thread](/docs/fundamentals/glossary#ui-thread).
+
+:::caution
+
+`setNativeProps` is an escape hatch for specific edge-cases.
+
+You should always reach for [`useAnimatedStyle`](docs/core/useAnimatedStyle) and [`useAnimatedProps`](docs/core/useAnimatedProps) first when animating styles or properties.
+
+:::
+
+## Reference
+
+```jsx
+import { setNativeProps } from 'react-native-reanimated';
+
+function App() {
+  const animatedRef = useAnimatedRef();
+
+  const handlePress = () => {
+    runOnUI(() => {
+      // highlight-start
+      setNativeProps(animatedRef, {
+        opacity: 0,
+      });
```

Comment:
Let's change this example to use `text` property, on `TextInput` can we? You really shouldn't opacity this way.
```suggestion
      setNativeProps(animatedRef, { text: '' });
```